xmodule/docs/decisions/0005-send-data-to-edx-submission.rst

```diff
@@ -0,0 +1,191 @@
+#########################################################
+Implementation to send student response to edx-submission
+#########################################################
+
+Status
+******
+
+Accepted.
+
+2025-03-06
+
+Context
+*******
+
+Traditionally, the Open edX platform has utilized the `XQueue` service to handle student assignment submissions. However, to leverage a more modern and efficient architecture, there is a need to transition these submissions to the `edx-submission` service. This transition aims to improve the scalability and maintainability of the submission handling process.
+
+Decision
+********
+
+The following changes have been implemented to redirect student submissions to the `edx-submission` service:
+
+1. **Introduction of `send_to_submission` Function in `xqueue_submission.py`:**
+   - **Functionality:** A new function, `send_to_submission`, has been developed to process student responses. It parses essential information such as `course_id`, `item_id`, `student_id`, and `max_score` from the submission data.
+   - **Integration with `edx-submission`:** The function utilizes the API provided by `edx-submission` to create a new submission record in the submissions database, ensuring that student responses are stored and processed accurately with the appropriate `max_score`.
+
+2. **Modification of `xqueue_interfaces.py`:**
+   - **Service Selection Logic:** The file `xmodule/capa/xqueue_interfaces.py` has been updated to include a condition that checks the state of the `send_to_submission_course.enable` *waffle flag*. If the flag is active for a specific course, student responses are directed to the `edx-submission` service via the `send_to_submission` function. If inactive, the submissions continue to be processed by `XQueue`.
+   - **Callback URL Construction:** The `construct_callback` method within the `XQueueService` class has been modified to generate the appropriate callback URL based on the state of the *waffle flag*. This ensures that evaluation results are returned correctly, whether processed by `edx-submission` or `XQueue`.
+
+3. **Temporary Waffle Flag Implementation:**
+   - **Definition:** A course-level waffle flag named `send_to_submission_course.enable` has been introduced. Administrators can set this flag via the Django admin interface, enabling or disabling the `edx-submission` functionality for specific courses without requiring code changes.
+   - **Deprecation Plan:** This *waffle flag* is a temporary measure to facilitate a smooth transition. Once the `edx-submission` service is fully adopted and validated, the flag will be deprecated, and all submissions will be processed exclusively through `edx-submission`.
+
+
+Configuration for Xqueue-watcher:
```

Comment:
@ormsbee - do you think this would be more appropriate somewhere in a `README` file rather than within an ADR?

Reply:
I think that it makes sense to eventually put it in the xqueue-watcher README once this is merged into master, but it's fine to stay in this ADR for now.